### PR TITLE
Tabular: Added unit tests for models

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Added unit tests for models
- FastAI NN unit tests are disabled at present since required dependencies are not installed during test runs on CI.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-764-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-764/1/index.html

File: tabular/tests/conftest.py [hunk 1]
```diff
@@ -19,3 +26,118 @@ def pytest_collection_modifyitems(config, items):
...
+            test_file=test_file,
+            name=name_inner,
+            url=url,
+        )
+
+        return train_data, test_data, dataset_info
+
+    @staticmethod
+    def dataset_info_dict():
+        dataset_dict = dict(
```
Comment (gradientsky) [inline tabular/tests/conftest.py:51]: nit: can be just a constant instead of method

File: tabular/tests/conftest.py [hunk 2]
```diff
@@ -19,3 +26,118 @@ def pytest_collection_modifyitems(config, items):
...
+        predictor.leaderboard(test_data, extra_info=True)
+        assert savedir == predictor.output_directory
+        if delete_directory:
+            shutil.rmtree(savedir, ignore_errors=True)  # Delete AutoGluon output directory to ensure runs' information has been removed.
+        return predictor
+
+    @staticmethod
+    def fit_dataset(train_data, fit_args, sample_size=None):
+        if sample_size is not None and sample_size < len(train_data):
+            train_data = train_data.sample(n=sample_size, random_state=0)
```
Comment (gradientsky) [inline tabular/tests/conftest.py:132]: should this be stratifed by target variable to ensure cases when in multiclass we could actually test binary classification?

**Review by gradientsky — CHANGES_REQUESTED** [review]

File: tabular/tests/conftest.py (hunk 2 above)
Comment (Innixma) [reply tabular/tests/conftest.py:132]: stratification won't actually solve that since we still drop classes with low count if eval_metric allows it (aka accuracy). Regardless, I don't think we need to worry about this edge case, since I am using large enough samples (1000) that it won't occur. The way to ensure it works would be via targeting log_loss + stratified sampling, but we probably don't need to go that far by having reasonable sample sizes.

File: tabular/tests/conftest.py (hunk 1 above)
Comment (Innixma) [reply tabular/tests/conftest.py:51]: Updated

File: tabular/tests/conftest.py [hunk 3]
```diff
@@ -19,3 +27,115 @@ def pytest_collection_modifyitems(config, items):
...
+            os.remove(zip_name)
+
+        train_data = task.Dataset(file_path=train_file_path)
+        test_data = task.Dataset(file_path=test_file_path)
+        return train_data, test_data
+
+
+class FitHelper:
+    @staticmethod
+    def fit_and_validate_dataset(dataset_name, fit_args, sample_size=1000, refit_full=True, delete_directory=True):
```
Comment (jwmueller) [inline tabular/tests/conftest.py:95]: do we want time_limits for potentially slow models?

**Comment by jwmueller** [comment]
Not sure if it's worthwhile, but we could also test individual models' HPO & bagging capabilities.

**Review by jwmueller — APPROVED** [review]

**Comment by szha** [comment]
Job PR-764-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-764/3/index.html

File: tabular/tests/conftest.py (hunk 3 above)
Comment (Innixma) [reply tabular/tests/conftest.py:95]: I considered it, but I think we should only do this if unit tests become slow, since the alternative is the test could fail because the model didn't have enough time to train and didn't return a result.

**Comment by Innixma** [comment]
> Not sure if it's worthwhile, but we could also test individual models' HPO & bagging capabilities.

I think this is reasonable, but we need to be careful about the time testing all of these things requires.

**Review by gradientsky — APPROVED** [review]
